packages/core/handle/handle.js

```diff
@@ -732,22 +732,105 @@ const adaptEvent = handle.adaptEvent = curry(function (adapter, handler) {
 	adaptEvent(
-		(...args) => {
-			let ev = adapter ? adapter(...args) : null;
+		(ev, ...args) => {
+			let customEventPayload = adapter ? adapter(ev, ...args) : null;
 
 			// Handle either no adapter or a non-object return from the adapter
-			if (!ev || typeof ev !== 'object') {
-				ev = {};
+			if (!customEventPayload || typeof customEventPayload !== 'object') {
+				customEventPayload = {};
 			}
 
-			ev.type = name;
+			customEventPayload.type = name;
+			if (typeof customEventPayload.preventDefault !== 'function' && typeof ev?.preventDefault === 'function') {
+				customEventPayload.preventDefault = ev.preventDefault.bind(ev);
+			}
+			if (typeof customEventPayload.stopPropagation !== 'function' && typeof ev?.stopPropagation === 'function') {
+				customEventPayload.stopPropagation = ev.stopPropagation.bind(ev);
+			}
 
-			return ev;
+			return customEventPayload;
 		},
 		forward(name)
 	)
 ).named('forwardCustom');
 
+/**
+ * Creates a handler that will forward the event to a function at `name` on `props` with capability
+ * to prevent default behavior. If the specified prop is `undefined` or is not a function, it is
+ * ignored. Returns `false` when `event.preventDefault()` has been called in a handler.
+ *
+ * If `adapter` is not specified, a new event payload will be generated with a `type` member with
+ * the `name` of the custom event. If `adapter` is specified, the `type` member is added to the
+ * value returned by `adapter`.
+ *
+ * The `adapter` function receives the same arguments as any handler. The value returned from
+ * `adapter` is passed as the first argument to `handler` with the remaining arguments kept the
+ * same. This is often useful to generate a custom event payload before forwarding on to a callback.
+ *
+ * Example:
+ * ```
+ * import {forwardCustomWithPrevent, handle} from '@enact/core/handle';
+ *
+ * // calls the onChange callback with the event: {type: 'onChange'}
+ * const forwardChangePreventDefault = handle(
+ *   forwardCustomWithPrevent('onChange'),
+ *   (ev) => console.log('default action', ev)
+ * );
+ *
+ * // calls the onChange callback with the event: {type: 'onChange', index}
+ * const forwardChangeWithIndexPreventDefault = handle(
+ *   forwardCustomWithPrevent('onChange', (ev, {index}) => ({index})),
+ *   (ev) => console.log('default action', ev)
+ * );
+ * ```
+ *
+ * @method   forwardCustomWithPrevent
+ * @param    {String}        name      Name of method on the `props`
+ * @param    {EventAdapter}  [adapter] Function to adapt the event payload
+ *
+ * @returns  {HandlerFunction}         Returns an [event handler]{@link core/handle.EventHandler}
+ *                                     (suitable for passing to handle or used directly within
+ *                                     `handlers` in [kind]{@link core/kind}) that will forward the
+ *                                     custom event and will return `false` if default action is prevented
```

Comment:
Since the return type is not bool, I am not sure if this part is correct.
It is also private api, and i can see what it means. so there is no better idea to me.
Please check it when merging.

Reply:
I agree. But I think this part makes sense since the subject is "an event handler".
Let's fix line 760 part.

Reply:
Done!